src/Storages/HDFS/ReadBufferFromHDFS.cpp

```diff
@@ -90,7 +91,7 @@ struct ReadBufferFromHDFS::ReadBufferFromHDFSImpl : public BufferWithOwnMemory<S
             num_bytes_to_read = internal_buffer.size();
         }
 
-        int bytes_read = hdfsRead(fs.get(), fin, internal_buffer.begin(), num_bytes_to_read);
+        int bytes_read = hdfsRead(fs.get(), fin, internal_buffer.begin(), safe_cast<int>(num_bytes_to_read));
```

Comment:
You are indeed right, it is very unlikely to get here > INT_MAX, however it can be a huge problem if here incorrect number of bytes will be read, hence `safe_cast`.